tests/test_nanopub.py

```diff
@@ -9,194 +9,211 @@
 
 client = NanopubClient(use_test_server=True)
 
-def test_nanopub_construction_with_bnode_introduced_concept():
-    """
-    Test Nanopub construction from assertion where a BNode is introduced as a concept.
-    """
-    test_uri = 'http://www.example.com/my-nanopub'
-    test_concept_uri = 'http://www.example.com/my-nanopub#DrBob'  # This nanopub introduced DrBob
-    assertion_rdf = rdflib.Graph()
-    assertion_rdf.add((rdflib.term.BNode('DrBob'),
-                       namespaces.HYCL.claims,
-                       rdflib.Literal('This is a test')))
-
-    nanopub = Nanopub.from_assertion(
-        assertion_rdf=assertion_rdf,
-        uri=rdflib.term.URIRef(test_uri),
-        introduces_concept=rdflib.term.BNode('DrBob'),
-        derived_from=rdflib.term.URIRef('http://www.example.com/another-nanopub')
-        )
-    assert str(nanopub.introduces_concept) == test_concept_uri
+TEST_ASSERTION = (namespaces.AUTHOR.DrBob, namespaces.HYCL.claims, rdflib.Literal('This is a test'))
+TEST_ORCID = rdflib.URIRef('http://example.org/pietje')
 
 
-@pytest.mark.flaky(max_runs=10)
-@skip_if_nanopub_server_unavailable
-def test_find_nanopubs_with_text():
-    """
-    Check that Nanopub text search is returning results for a few common search terms
-    """
-    searches = ['test', 'US']
+def _get_mock_profile():
+    mock_profile = mock.MagicMock()
+    mock_profile.get_orcid.return_value = TEST_ORCID
+
+    return mock_profile
 
-    for search in searches:
-        results = client.find_nanopubs_with_text(search)
-        assert len(results) > 0
 
-    assert len(client.find_nanopubs_with_text('')) == 0
+# Created a class so profile can be mocked for all tests at once
+@mock.patch('nanopub.nanopub.profile', _get_mock_profile())
+class NanopubTest(TestCase):
```

Comment:
Do we need to subclass `TestCase` here? If not maybe leave it out?

Reply:
```suggestion
class TestNanopub:
```
I would also say this is more common, but curious what you think!

Reply:
Also now that this is in a class, it would be good to make a distinction between tests for Nanopub class and NanopubClient. (i.e, TestNanopub & TestNanopubClient)

Reply:
yeah if it were up to me the nanopub client would be in a separate module called NanopubClient.
In the `__init__.py` we could then say
```
from NanopubClient import *
```

And then I would put the tests in a separate module as well. What do you think?

Reply:
By the way, I tried to remove the subclassing stuff but pytest is unable to find the test if I do not subclass from testcase so I left it in. Let me know if you have a solution for that

Reply:
Yeah, I agree to split client and Nanopub class. I will pick it up in a separate PR.

And see my suggestion for renaming `NanopubTest` to `TestNanopub`. Then pytest will be able to find it.

Reply:
it works